x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_schema.tsx

```diff
@@ -143,12 +144,23 @@ const SchemaFieldDetails: React.FC<{ schemaField: SchemaField }> = ({ schemaFiel
 
 export const EngineSchema: React.FC = () => {
   const { engineName } = useValues(EngineIndicesLogic);
+  const [onlyShowConflicts, setOnlyShowConflicts] = useState<boolean>(false);
   const { isLoadingEngineSchema, schemaFields } = useValues(EngineViewLogic);
   const { fetchEngineSchema } = useActions(EngineViewLogic);
   const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, JSX.Element>>(
     {}
   );
 
+  const toggleOnlyShowConflicts = useCallback(() => {
+    setOnlyShowConflicts(!onlyShowConflicts);
+    setItemIdToExpandedRowMap({});
+  }, [setItemIdToExpandedRowMap, onlyShowConflicts]);
```

Comment:
question: `setItemIdToExpandedRowMap` should never change, right? Why do we need that as a dependency for this callback?

Reply:
✅ https://github.com/elastic/kibana/pull/154149/commits/254c9ae52fbf0c8e931896e8d786a7b7a3362aee